src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -1405,6 +1411,33 @@ private RegisteredUserDTO convertUserDOToUserDTO(final RegisteredUser user) {
         return userDTO;
```

Comment:
What you have done is sensible and should speed things up by hitting the database less. But, the logic is now duplicated between `ConvertUserDOToUserDTO` and `converUserDOsToUserDTOs`. Changing the body of `convertUserDOToDTO` to just call the other with a list of size one will eliminate this duplication and avoid future bugs where we update one and not the other.

Same goes for `getAuthenticationProvidersByUsers` and `getSegueAccountExistenceByUsers`.

Reply:
I agree, but this might take a while as I don't know how widespread these calls are.